resources/js/main.js

```diff
@@ -0,0 +1,267 @@
+pkp.registry.storeExtend("fileManager_SUBMISSION_FILES", (piniaContext) => {
+  const fileStore = piniaContext.store;
+  
+  // Only extend for submissions with CODECHECK
+  const workflowStore = pkp.registry.getPiniaStore("workflow");
+  const submission = workflowStore?.submission;
+  
+  if (!submission?.codecheckOptIn) {
+    return;
+  }
+
+  fileStore.extender.extendFn("getColumns", (columns, args) => {
+    const newColumns = [...columns];
+    
+    const { useLocalize } = pkp.modules.useLocalize;
+    const { t } = useLocalize();
+
+    // Add CODECHECK status column before actions
+    newColumns.splice(newColumns.length - 1, 0, {
+      header: t("plugins.generic.codecheck.codecheckStatus"),
+      component: "CodecheckFileStatus",
+      props: {},
+    });
+
+    return newColumns;
+  });
+
+  fileStore.extender.extendFn("getItemActions", (originalResult, args) => {
+    if (args.file) {
+      return [
+        ...originalResult,
+        {
+          label: "Mark as CODECHECK Output",
+          name: "markCodecheckOutput",
+          icon: "CheckCircle",
+          actionFn: ({ file }) => {
+            const { useModal } = pkp.modules.useModal;
+            const { useLocalize } = pkp.modules.useLocalize;
+
+            const { openDialog } = useModal();
+            const { localize } = useLocalize();
+
+            openDialog({
+              title: "Mark as CODECHECK Output",
+              message: `Do you want to mark "${localize(file.name)}" as a CODECHECK output file?`,
+              actions: [
+                {
+                  label: "Yes",
+                  isPrimary: true,
+                  callback: (close) => {
+                    // TODO: Implement marking file as CODECHECK output
+                    console.log("Marking file as CODECHECK output:", file);
+                    close();
+                  },
+                },
+                {
+                  label: "No",
+                  callback: (close) => {
+                    close();
+                  },
+                },
+              ],
+            });
+          },
+        },
+      ];
+    }
+    return originalResult;
+  });
+});
+
+// Mount Vue components for traditional form fields
+window.addEventListener('DOMContentLoaded', () => {
+  setTimeout(() => {
+    mountCodecheckVueComponents();
+  }, 1000);
+});
+
+function mountCodecheckVueComponents() {
+  // Mount manifest files component
+  const manifestContainer = document.querySelector('textarea[name="manifestFiles"]')?.parentElement;
+  if (manifestContainer) {
+    const textarea = manifestContainer.querySelector('textarea');
+    const vueDiv = document.createElement('div');
+    manifestContainer.insertBefore(vueDiv, textarea);
+    textarea.style.display = 'none';
+    
+    createApp(CodecheckManifestFiles, {
+      name: 'manifestFiles',
+      label: 'Expected Output Files',
+      description: 'List the main figures, tables, and results',
```

Comment:
Expand description:

"List the main figures, tables, and results that the codechecker should consider for their evaluation."

Also, this description needs to be translated! Find out how you can load strings from `locale.po` into Vue.js components...